### DISCUSSION TITLE
ENVELOPE ENHANCEMENTS (delay, hold, repeat)

### DISCUSSION THREAD

**Discussion by worker-gray** [SynthstromAudible/DelugeFirmware#579] [Ideas]
What would help with sound design is if there were modifiers to enhance the envelope to include delay [as part of the attack], hold [as part of decay] and repeat [as part of the release] to further sculpt the sound.
Delay would put a 'pause' on the effects of the envelope before the attack-- could be specified in milliseconds, seconds or clock divisions.
Hold would keep the max attack level -- again, with clock divisions
Repeat is self explanatory- perhaps clock divisions would quantize the envelope to fit within the specified divisions, or 'vanilla' would just repeat the entire envelope at termination. Almost a third LFO

**Comment by ok-reza** [comment]
how do you think the user would find these enhancements given the lack of shortcut space we have? i agree it would be nice to have, just implementing in a smart/consistent/discoverable way to really important for it to have life beyond the idea.

**Reply by worker-gray** [reply]
I would 'nest' them under the already existing shortcuts for the envelope parameters--
Double tap. first tap, primary parameters, second tap, secondary parameters.
You can only access the 'delay' function after accessing the 'attack' function, 'hold' after 'decay' and finally 'repeat' after 'release'.
The third tap would cycle back to primary, and so on.

**Reply by ok-reza** [reply]
i think that's a solid idea, it's also consistent with what's been implemented with unison stereo width (clicking into unison number reveals unison stereo width). my only concern is that some people might finish editing a parameter by clicking in, if this is implemented in a high traffic area (env1 adsr) clicking will never mean 'done and exit', it will just cycle between to two options. i'm pretty much fine with it since i almost always hit mute/launch to exit out of most things, but some confusion may arise and should be expected to some degree.

i think it should be:
attack > delay
**decay > nothing
sustain > hold**
release > repeat
instead. i feel like hold mirrors sustain and makes more sense, what do you think?

**Reply by worker-gray** [reply]
Absolutely. That works for me.
As far as confusing the users, maybe it could be an option that they can turn off in community settings OR it would just display what page you're on the screen (whichever type you have) -- maybe with a 'DOT' to indicate extended parameters on the 7 segment.
